proversity_reports_script/report_backend/time_spent_per_user_report.py

```diff
@@ -0,0 +1,292 @@
+"""
+Time spent per user report backend.
+"""
+import csv
+import functools
+import os
+from datetime import datetime
+
+import boto3
+
+from proversity_reports_script.google_apis.sheets_api import update_sheets_data
+from proversity_reports_script.report_backend.base import AbstractBaseReportBackend
+
+
+class TimeSpentPerUserReportBackend(AbstractBaseReportBackend):
+    """
+    Backend for time spent per user report.
+    """
+
+    def __init__(self, *args, **kwargs):
+        extra_data = kwargs.get('extra_data', {})
+        self.bucket_name = extra_data.get('BUCKET_NAME', '')
+
+        super(TimeSpentPerUserReportBackend, self).__init__(extra_data.get('SPREADSHEET_DATA', {}))
+
+    def generate_report(self, json_report_data):
+        """
+        Main logic to generate the report.
+        """
+        self.json_report_to_csv(json_report_data)
+
+    def json_report_to_csv(self, json_report_data):
+        """
+        Process json data to convert into csv format.
+
+        Args:
+            json_report_data: Json representacion containing time spent report data.
+        """
+        report_data = json_report_data.get('result', {})
+
+        if not report_data:
+            print('No report data...')
+            exit()
+
+        for course_key, course_data in report_data.items():
+            file_name = '{}'.format(course_key)
+
+            if not course_data:
+                continue
+
+            try:
+                time_spent_per_user_report_data = build_time_spent_report_data(course_data)
+
+                self.create_csv_file(
+                    file_name=file_name,
+                    body_dict=time_spent_per_user_report_data,
+                    headers=time_spent_per_user_report_data[0].keys(),
+                    course_id=course_key,
+                    spreadsheet_id=self.spreadsheet_data.get('time_spent_sheet_id_{}'.format(course_key)),
+                    spreadsheet_range_name='Sheet1',
+                )
+
+                time_spent_per_team_report_data = grouping_report_data_by_key(course_data, 'user_teams')
+
+                self.create_csv_file(
+                    file_name='{}-teams'.format(file_name),
+                    body_dict=time_spent_per_team_report_data,
+                    headers=time_spent_per_team_report_data[0].keys(),
+                    course_id=course_key,
+                    spreadsheet_id=self.spreadsheet_data.get('time_spent_sheet_id_{}'.format(course_key)),
+                    spreadsheet_range_name='Sheet2',
+                    file_name_prefix='teams-',
+                )
+
+                time_spent_per_cohort_report_data = grouping_report_data_by_key(course_data, 'user_cohort')
+
+                self.create_csv_file(
+                    file_name='{}-cohorts'.format(file_name),
+                    body_dict=time_spent_per_cohort_report_data,
+                    headers=time_spent_per_cohort_report_data[0].keys(),
```

Comment:
you can get this inside the method by using the body_dict

Reply:
Addressed. @andrey-canon